pkg/bitcoin/hd_test.go

```diff
@@ -0,0 +1,140 @@
+package bitcoin
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/btcsuite/btcutil/hdkeychain"
+	pb "github.com/ledgerhq/lama-bitcoin-svc/pb/v1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// h indicates the BIP32 harden bit, equivalent to 2^31.
+var h uint32 = 0x80000000
+
+func TestDeriveExtendedKey(t *testing.T) {
```

Comment:
Can you add more error cases ? 
Empty array for instance ? Does it trigger properly _InvalidArgument Code = 3_ ? 

Can you specify here result expected when we have array with more than 2 elems ?

Reply:
> Empty array for instance ? Does it trigger properly InvalidArgument Code = 3 ?

It shouldn't return an error IMO. Passing an empty array is akin to no derivation, and the method will simply return the public key and chain code of the extended key passed as input. I can add a unit-test for this.

> Can you specify here result expected when we have array with more than 2 elems ?

Array of all sizes should work without any issues.

The `mainnet derive from root` test covers this remark. I intentionally picked an example from the official test vectors that let me derive 5 HD levels, with unhardened indexes since we're dealing with public keys only.

Reply:
Test proof is the way :) 

> The mainnet derive from root test covers this remark. I intentionally picked an example from the official test vectors that let me derive 5 HD levels, with unhardened indexes since we're dealing with public keys only.

✔️  🚀

Reply:
I just pushed a unit-test with an empty array. You'll have to trust me that it passes since there is no CI. 😅